qutebrowser/browser/hints.py

```diff
@@ -130,7 +137,23 @@ def _move_to_elem(self):
     def cleanup(self):
         """Clean up this element and hide it."""
         self.hide()
-        self.deleteLater()
+        self.setParent(None)
+        self._context.tab.contents_size_changed.disconnect(self._move_to_elem)
+        HintLabel._object_pool.append(self)
+
+    @staticmethod
+    def create(elem, context):
+        """Returns a HintLabel object with the parameters (elem, context).
+
+        Try to reuse an object in the object pool if possible. If not possible,
+        create a new one.
+        """
+        try:
+            self = HintLabel._object_pool.pop() # type: HintLabel
+        except IndexError:
+            return HintLabel(elem, context)
+        self.__init__(elem, context, first_time=False)
```

Comment:
After I noted that the order of some statements are important, I think it would be better if it's possible to order the statements arbitrarily (the older approach forces the reinit statements to be continuous) About weird side effect, no as long as code correctly pass the `first_time` parameter.

Reply:
I think this thread sums it up https://stackoverflow.com/a/13091370

Reply:
What would be the best solution:

1. 

```
def __init__(self, elem, context):
    self.reset(elem, context, first_time=True)

def reset(self, elem, context, first_time):
    # logic for reset here

def create(...):
    ...
    label.reset(elem, context, first_time=False)
```

Drawback: Passing `first_time=False` to `reset` doesn't make much sense.

2. 

```
def __init__(self, elem, context):
    # code that works like `first_time=True`

def reset(self, elem, context, first_time):
    # code that works like `first_time=False`
```

Drawback: Code duplication in `__init__` and `reset`.

3.

```
def __init__(self, elem, context):
    # ...
    self.reset(elem, context)
    # ...

def reset(self, elem, context, first_time):
    # ...
```

Drawback: The initialization statements needs to be reordered such that the code in `reset` is executed in a contiguous block. Initially I wrongly ordered it (it's important that `show` is executed last), therefore causes a bug.

Reply:
I think I prefer 3. - it seems quite usual for statements in `__init__` to have a defined order anyways.